app-next/app/meals/[id]/page.js

```
@@ -0,0 +1,167 @@
+// app/meals/[id]/page.js
+import Image from "next/image";
+import { MealFormsWrapper } from "../../../components/MealFormsWrapper/MealFormsWrapper";
+import "./MealDetails.css";
+
+export async function generateStaticParams() {
+  const backendApiUrl = process.env.NEXT_PUBLIC_BACKEND_API_URL;
+  let meals = [];
+
+  try {
+    const response = await fetch(`${backendApiUrl}/meals`, {
+      headers: {
+        "Cache-Control": "no-cache",
+        Pragma: "no-cache",
+        Expires: "0",
+      },
+      cache: "no-store",
+    });
+
+    if (!response.ok) {
+      return [];
+    }
+
+    const data = await response.json();
+    meals = data.meals || data;
+  } catch (error) {
+    return [];
+  }
+
+  const params = meals.map((meal) => ({
+    id: meal.id.toString(),
+  }));
+  return params;
+}
+
+async function getMealDetails(id) {
+  try {
+    const backendApiUrl = process.env.NEXT_PUBLIC_BACKEND_API_URL;
+    const response = await fetch(`${backendApiUrl}/meals/${id}`, {
+      cache: "no-store",
+    });
+
+    if (!response.ok) {
+      return null;
```

In modern programming, it is not recommended to return null to indicate an error. Think about how the null will be treated in the calling function, etc. 

Instead, you can throw the error back to the calling function, and handle it accordingly over there.